pandas/_libs/tslibs/period.pyx

```diff
@@ -2574,16 +2574,15 @@ class Period(_Period):
                 value = str(value)
             value = value.upper()
             dt, reso = parse_time_string(value, freq)
-            try:
-                ts = Timestamp(value)
-            except ValueError:
-                nanosecond = 0
-            else:
-                nanosecond = ts.nanosecond
-                if nanosecond != 0:
-                    reso = "nanosecond"
+            if reso == "nanosecond":
+                nanosecond = dt.nanosecond
+
             if dt is NaT:
                 ordinal = NPY_NAT
+                # Doesn't matter what this is, we just need to have it
```

Comment:
Instead of assigning a random value here would it make more sense to just not have it fail later on?